extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcSession.java

```diff
@@ -16,12 +17,21 @@ public interface OidcSession {
     String getTenantId();
 
     /**
-     * Return an {@linkplain:Instant} indicating how long will it take for the current session to expire.
+     * Return an {@linkplain Instant} representing the current session's expiration time
+     * which is a number of seconds from the epoch of 1970-01-01T0:0:0Z.
      *
-     * @return
+     * @return Instant
      */
     Instant expiresIn();
 
+    /**
+     * Return a {@linkplain Duration} indicating how long will it take for the current session to expire
+     * starting from this method's invocation time.
+     *
+     * @return Duration
+     */
+    Duration getExpiresInDuration();
```

Comment:
This is really less than ideal. I would have expected something like `getExpirationTime` returning an Instant and `expiresIn` returning a duration.
Given this previous method is of not much use right now, I think we should break things now, rather than living with some badly named methods forever.

Reply:
If breaking the API is an option here, then I agree with @gsmet, but I'd name it `expirationTime()` instead 😉

Reply:
Well, we somehow break the API as we won't return the same thing that it used to be. If people are using the existing API, things will get messy as the meaning has changed.
So I would prefer if we broke things cleanly.

Agreed on `expirationTime()` being better since apparently getters were not used for the rest.

Reply:
@gsmet Fix to the original API is not a breaking change but a bug fix, I won't be removing it but I'm fine with deprecating it as I've already had some pushback for just breaking things in a supported extension.

So we'll have 
```
@Deprecated
@use expirationTime();
Instant expiresIn();

// new method
Instant expirationTime();

// new method - anything better than expiresInPeriod ? It is not ideal but IMHO it works, the docs make it clear what it is for
Duration expiresInPeriod();

```

Reply:
To be honest I think `expiresInDuration` is also good enough, I'd not make a big deal out of it

Reply:
Actually `on` is for dates, so it would be used in the expirationTime  https://continuingstudies.uvic.ca/elc/studyzone/490/grammar/prepositions-time-in-at-on

Reply:
may be `Duration validFor()`

Reply:
@sberyozkin it is a breaking change as the meaning is different. If people were using the value before to do something useful, it will be broken as you will return something with a totally different meaning.
That's why I would prefer if we break the API. That way, you know that the meaning has changed.

Reply:
Well. I don't. I think `expiresAt` for instance and `expiresIn` for duration would be better candidates. But you seem to have a strong opinion about it so I will let you decide and live with it :).

Reply:
And `validFor` is indeed `less worse` then the previous proposals :).

Reply:
@gsmet I agree with you, and I do appreciate you keeping an eye, but I'm just trying to workaround about the fact I might need to keep the current `expiresIn`, but I'm not sure now this interface made it to the product, I'll check, if not I'll be happy to get rid of the current `expiresIn`

Reply:
@gsmet Sadly it was committed to 2.7.0.CR1, did it for Stephane but this expiry thing was not even requested, just added it in a rush in addition to `logout()`, so I guess the proposed strategy where we try to return something meaningful from this currently badly implemented, explain it in the docs but deprecate at the same time, and then have as you propose `Instance expirationTime` (saw you agreeing on it above) and I guess then `Duration validFor()` may work.

But I'll ping a few more colleagues to discuss, thanks

Reply:
`expiresAt` is probably a bit better than `expirationTime`.

Reply:
@gsmet Sure, I've just checked, for the token gen API, it is [expiresAt for Instant](https://github.com/smallrye/smallrye-jwt/blob/main/implementation/jwt-build/src/main/java/io/smallrye/jwt/build/JwtClaimsBuilder.java#L105) and [expiresIn for Duration](https://github.com/smallrye/smallrye-jwt/blob/main/implementation/jwt-build/src/main/java/io/smallrye/jwt/build/JwtClaimsBuilder.java#L125) :-). I'm about to push the update, but if you'd like then we can avoid adding `Duration validFor()` just yet - but do it in `2.13.0.CR1` where we can delete (though not 100% sure about it just yet) the deprecated method and add `Duration expiresIn()`

Reply:
@gsmet Have a look please, like I said, we can postpone adding `Duration validFor` unless it would imply we'd have to wait for a long time before it can be added